api/src/wfl/api/handlers.clj

```diff
@@ -118,65 +92,74 @@
+                (postgres/load-workload-for-id tx id))
+              workload))]
+    (fail-on-error
+      (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+        (succeed
+          (mapv (partial go! tx)
+            (if-let [uuid (-> request :parameters :query :uuid)]
+              (if-let [workload (postgres/load-workload-for-uuid tx uuid)]
+                [workload]
+                (throw (ex-info "No such workload" {:uuid uuid})))
+              (postgres/load-workloads tx))))))))
 
 (defn post-start
   "Start the workloads with UUIDs in REQUEST."
   [request]
-  (wfl.api.handlers/print-handler-error
-   (let [uuids (-> request :parameters :body distinct)]
-     (logr/infof "post-start endpoint called: uuids=%s" uuids)
-     (letfn [(q [[left right]] (fn [it] (str left it right)))]
-       (let [db-config (postgres/wfl-db-config)
-             db-conn (jdbc/get-connection db-config)
-             query (->> (repeat (count uuids) "?")
-                        (str/join ",") ((q "()"))
-                        (format "SELECT * FROM workload WHERE uuid in %s"))
-             ps (jdbc/prepare-statement db-conn query)]
-         (doseq [[i uuid] (map-indexed vector uuids)] (.setString ps (+ i 1) (:uuid uuid)))
-         (jdbc/with-db-transaction [tx db-config]
-           (->> ps
-                (jdbc/query tx)
-                (run! (partial start-workload! tx)))
-           (->> uuids
-                (mapv (partial postgres/get-workload-for-uuid tx))
-                succeed)))))))
-
-(def post-exec
+  (letfn [(go! [tx {:keys [uuid]}]
+            (if-let [workload (postgres/load-workload-for-uuid tx uuid)]
+              (if-not (:started workload)
+                (do
+                  (logr/infof "starting workload %s" uuid)
+                  (start-workload! tx workload)
+                  (postgres/load-workload-for-id tx (:id workload)))
+                workload)
+              (throw (ex-info "No such workload" {:uuid uuid}))))]
+    (fail-on-error
+      (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+        (let [uuids (-> request :parameters :body distinct)]
+          (logr/infof "post-start endpoint called: uuids=%s" uuids)
+          (succeed (mapv #(go! tx %) uuids)))))))
+
+(defn post-exec
   "Create and start workload described in BODY of REQUEST"
-  (let [pack   (fn [response] {:parameters {:body [(:body response)]}})
-        unpack (comp first :body)]
-    (comp succeed unpack post-start pack post-create)))
+  [request]
+  (letfn [(create! [tx workload-request]
+            (let [{:keys [uuid]} (add-workload! tx workload-request)]
+              (log/infof "created workload %s" uuid)
+              (postgres/load-workload-for-uuid tx uuid)))
+          (start! [tx {:keys [id uuid] :as workload}]
+            (log/infof "starting workload %s" uuid)
+            (start-workload! tx workload)
+            (postgres/load-workload-for-id tx id))
```

Comment:
I think you want (at least) a `(defn start-workload …)`.

Reply:
For posterity:
There are two `start!` letfns which look almost identical.
The plan is that this will be absorbed into the `workload` "class"